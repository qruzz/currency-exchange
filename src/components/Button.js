import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * This function returns a button component to be rendered. The function
 * takes the props objects as the only parameter and styles the button
 * accordingly. The button will fire an onclick function, if passed through.
 * @param	{object}	props	The props for the component
 * @returns	{jsx}				Returns the button component
 */
function Button({ onClick, title, color, type, active }) {
	return (
		<Wrapper
			color={color}
			active={active}
			type={type}
			onClick={active ? onClick : null}
		>
			<Title>{title}</Title>
		</Wrapper>
	);
}

// Verify the passed through Props
Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	title: PropTypes.string.isRequired,
	color: PropTypes.string,
	type: PropTypes.string,
	active: PropTypes.bool,
};

// Define the default PropTypes
Button.defaultProps = {
	color: '#EB008D',
	type: 'primary',
	active: false,
};

const Wrapper = styled.button`
	width: ${props => props.type === 'primary' ? '100%' : '20%'};
	height: 6vh;
	display: flex;
	align-items: center;
	justify-content: center;
	border-radius: calc(12vh / 2);
	outline: none;
	background: ${props => props.color};
	opacity: ${props => props.active ? 1 : 0.4};
	cursor: ${props => props.active ? 'pointer' : 'default' };
	margin: 5% 0;

	@media (min-width: 700px) {
		width: 30%;
	}
`;

const Title = styled.span`
	color: #FFFFFF;
`;

export default Button;
