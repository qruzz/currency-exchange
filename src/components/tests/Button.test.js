import React from 'react';
import { shallow, expect } from '../../enzyme';

import Button from '../Button';

describe('Button tests', () => {
	it('renders a button', () => {
		const items = ['one', 'two', 'three'];
		const wrapper = shallow(<Button items={items} />);

		// Expect the wrapper object to be defined
		expect(wrapper.find('.list-items')).toBeDefined();
		expect(wrapper.find('.item')).toHaveLength(items.length);
	});

	it('renders a list item', () => {
		const items = ['Thor', 'Loki'];
		const wrapper = shallow(<List items={items} />);

		// Check if an element in the Component exists
		expect(wrapper.contains(<li key='Thor' className="item">Thor</li>)).toBeTruthy();
	});

	it('renders correct text in item', () => {
		const items = ['John', 'James', 'Luke'];
		const wrapper = shallow(<List items={items} />);

		// Expect the child of the first item to be an array
		expect(wrapper.find('.item').get(0).props.children).toEqual('John');
	});
});
