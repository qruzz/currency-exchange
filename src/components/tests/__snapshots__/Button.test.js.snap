// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders a button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <Button
      active={false}
      color="#EB008D"
      type="primary"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "children": <ForwardRef />,
        "color": "#EB008D",
        "onClick": null,
        "type": "primary",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
	color: #FFFFFF;
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
	width: ",
            [Function],
            ";
	height: 6vh;
	display: flex;
	align-items: center;
	justify-content: center;
	border-radius: calc(12vh / 2);
	outline: none;
	background: ",
            [Function],
            ";
	opacity: ",
            [Function],
            ";
	cursor: ",
            [Function],
            ";
	margin: 5% 0;

	@media (min-width: 700px) {
		width: 30%;
	}
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": <ForwardRef />,
          "color": "#EB008D",
          "onClick": null,
          "type": "primary",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
	color: #FFFFFF;
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
	width: ",
              [Function],
              ";
	height: 6vh;
	display: flex;
	align-items: center;
	justify-content: center;
	border-radius: calc(12vh / 2);
	outline: none;
	background: ",
              [Function],
              ";
	opacity: ",
              [Function],
              ";
	cursor: ",
              [Function],
              ";
	margin: 5% 0;

	@media (min-width: 700px) {
		width: 30%;
	}
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "children": <ForwardRef />,
        "color": "#EB008D",
        "onClick": null,
        "type": "primary",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
	color: #FFFFFF;
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
	width: ",
            [Function],
            ";
	height: 6vh;
	display: flex;
	align-items: center;
	justify-content: center;
	border-radius: calc(12vh / 2);
	outline: none;
	background: ",
            [Function],
            ";
	opacity: ",
            [Function],
            ";
	cursor: ",
            [Function],
            ";
	margin: 5% 0;

	@media (min-width: 700px) {
		width: 30%;
	}
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
}
`;
